Remove dispatch_async now that APS is working

LumioseSil opened the pull request:
#### 00d8e875174dae409eb44ff6046eade899789649
<pre>
Remove dispatch_async now that APS is working
<a href="https://bugs.webkit.org/show_bug.cgi?id=251806">https://bugs.webkit.org/show_bug.cgi?id=251806</a>

Reviewed by Ben Nham.

Now that APS is working again, the dispatch_async can be removed.

* Source/WebKit/webpushd/ApplePushServiceConnection.mm:
  (didReceiveIncomingMessage): Remove dispatch_async now that APS is
  working as of macOS Ventura.

Canonical link: <a href="https://commits.webkit.org/265201@main">https://commits.webkit.org/265201@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/68f7b3c5ebb5f02fade815300f78f71b276466c4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/9965 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/10211 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/10463 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/11616 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/9652 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/12199 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/10164 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/11616 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/10120 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/12199 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/10463 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/12000 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/12199 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/10463 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/12000 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/12199 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/10463 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/12000 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/9684 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/10164 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/8843 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/10463 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2451 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/13072 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/9462 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/webpushd/ApplePushServiceConnection.mm:
```diff
@@ -114,11 +114,8 @@ - (void)connection:(APSConnection *)connection didReceiveIncomingMessage:(APSInc
         if (!weakThis)
             return;
 
-        // FIXME: remove this once APS stops calling back on the wrong queue (rdar://88100275).
```
What OS versions do we need this code to work on? Not sure if webpushd gets updated by the Safari installer.

AZero13 replied:
Big Sur, Monterey, Ventura, the usual suspects.

aproskuryakov replied:
I don't think that rdar://88100275 is fixed on all of those.

AZero13 replied:
Should be.

aproskuryakov replied:
This is not what I see in rdar://88100275, so please share the evidence of it having been fixed early enough.

cdumez replied:
This file is guarded by APPLE_PUSH_SERVICE_URL_TOKEN_SUPPORT, which is enabled on macOS 13+ (Ventura) and iOS 16+. Based on rdar://88100275, this was was indeed fixed in macOS 13+  and iOS 16+ so we probably don't need the workaround anymore.

I'll let @beidson double-check my findings though.

bnham replied:
This was needed to fix an issue in pre-GM Ventura. From what I can tell it is fixed in GM Ventura so I'll approve it (as webpushd only runs starting in Ventura even on downlevels).

AZero13 replied:
Thank you!

LumioseSil committed:
Remove dispatch_async now that APS is working
https://bugs.webkit.org/show_bug.cgi?id=251806

Reviewed by Ben Nham.

Now that APS is working again, the dispatch_async can be removed.

* Source/WebKit/webpushd/ApplePushServiceConnection.mm:
  (didReceiveIncomingMessage): Remove dispatch_async now that APS is
  working as of macOS Ventura.

Canonical link: https://commits.webkit.org/265201@main
